### DISCUSSION TITLE
Module '"next-intl/server"' has no exported member 'getRequestConfig'

### DISCUSSION THREAD

**Discussion by nicitaacom** [amannn/next-intl#541] [Q&A]
### Description

I use Next 13.5.3
I follow this guide - https://next-intl-docs.vercel.app/docs/getting-started/app-router-server-components
When I paste this to my i18n.ts (in root directory)
```ts
import {getRequestConfig} from 'next-intl/server';
 
export default getRequestConfig(async ({locale}) => ({
  messages: (await import(`./messages/${locale}.json`)).default
}));
```

I got this error
`Module '"next-intl/server"' has no exported member 'getRequestConfig'`

### Mandatory reproduction URL (CodeSandbox or GitHub repository)

https://codesandbox.io/p/sandbox/i18n-thzc9v

### Reproduction description

Steps to reproduce:
1. Open codesandbox
2. Click on VS conde in top right corner
3. See error in i18n.ts

### Expected behaviour

I wait no error
But got an error

**Comment by amannn** [comment]
Any chance `next-intl` got installed incorrectly?

When I run your example I also get this error:

<img width="627" alt="Screenshot 2023-10-02 at 16 59 21" src="https://github.com/amannn/next-intl/assets/4038316/a6ca2e9a-fad8-40d1-9a17-de487c1e4670">

Seems like your `middleware.ts` example is empty.

I'll move this to a discussion since it seems to be a usage question.

**Reply by nicitaacom** [reply]
Here - https://codesandbox.io/p/sandbox/i18n-thzc9v?file=%2Fi18n.ts%3A5%2C5

I pasted code in middleware.ts


Ok in this case error gone

You have wrong docs
It should be like this

![image](https://github.com/amannn/next-intl/assets/39565703/76dca681-a9d5-4245-ba22-09dbbd042ea9)

**Comment by nicitaacom** [comment]
> Any chance `next-intl` got installed incorrectly?
> 
> When I run your example I also get this error:
> 
> <img alt="Screenshot 2023-10-02 at 16 59 21" width="627" src="https://user-images.githubusercontent.com/4038316/272007164-a6ca2e9a-fad8-40d1-9a17-de487c1e4670.png">
> Seems like your `middleware.ts` example is empty.
> 
> I'll move this to a discussion since it seems to be a usage question.


![image](https://github.com/amannn/next-intl/assets/39565703/eb4b4bcc-c137-4c89-97d1-2f5b2a3cdf8c)


cur - what is that?

**Comment by nicitaacom** [comment]
I got 404 error

What I'm doing wrong?

codesandbox - https://codesandbox.io/p/sandbox/i18n-thzc9v?file=%2Fi18n.ts%3A5%2C5

**Comment by nicitaacom** [comment]
Also I found this issue in docs
![image](https://github.com/amannn/next-intl/assets/39565703/b5c058b1-27be-4a84-aa41-ccd3efdd2bce)

**Comment by MarlonWiss2212** [comment]
I needed to install the beta to fix it 
`npm install next-intl@3.0.0-beta.19`

**Reply by nicitaacom** [reply]
Send me example please
https://github.com/i18next/next-i18next/discussions/2223

**Comment by huypham1411** [comment]
I also got this issue along with this https://github.com/amannn/next-intl/discussions/640 when upgrading the package to the latest version but I cannot find any solution to solve this issue. This issue didn't occur in the `next-intl@3.0.0-beta.19` but the project still can run normally and build without any error

**Reply by nicitaacom** [reply]
Please tell me how you got it
Today I tried to fork official example and got the error during build process
![image](https://github.com/amannn/next-intl/assets/39565703/82f90314-f852-4d65-92be-bdd47eb28339)


Please send me github repositoty where you have no error

**Reply by huypham1411** [reply]
![image](https://github.com/amannn/next-intl/assets/54835897/dfc8ad7b-893a-4ac7-bdb4-58bb71c32eac)
here is my project structure, you can also take a look at this video https://www.youtube.com/watch?v=pKolfZFW3gs, hope it helps you solve the issue

**Reply by nicitaacom** [reply]
He use JavaScript - so he have no that error

Also he didn't deployed it - just run project in development I can as well

Its 'pseudo' guide - a lot of that on youtube (I noticed) when you deploy something with next-auth or clerk it doesn't work as well

If you figure it out - please create step by step guide how to do it and pushish on YouTube (with deploy and showing that if works after deploy)
